### PR TITLE
Reduce optional allocations

### PR DESCRIPTION
**Pull request by pzartem**
- Reduce allocations in optional fields.
- Fixing `bitpack` and it's gen so now it has an allocation free interface. RLE is also can reuse a buffer.
- Updating test with a new generator.

```
benchmark               iter      time/iter      bytes alloc              allocs
---------               ----      ---------      -----------              ------
// master
BenchmarkWrite/opt-8      22   486.70 ms/op    89216541 B/op   4308470 allocs/op
// PR
BenchmarkWrite/opt-8      28   440.07 ms/op    69019114 B/op    133440 allocs/op
```

### CODE REVIEW

**Comment by pzartem** [comment]
@lev112 Can you try to run benchmark on your machine?

**Comment by lev112** [comment]
here is the benchmark on my machine, looking good
```
benchmark               iter      time/iter      bytes alloc              allocs
---------               ----      ---------      -----------              ------
// master
BenchmarkWrite/opt-8      22   486.70 ms/op    89216541 B/op   4308470 allocs/op
// PR
BenchmarkWrite/opt-8      28   440.07 ms/op    69019114 B/op    133440 allocs/op
```

File: performance/parquet_performance_test.go [hunk 1]
```diff
@@ -74,7 +75,10 @@ func benchmarkParquet(b *testing.B, data []message.Message, buf *bytes.Buffer, g
 func BenchmarkWrite(b *testing.B) {
 	data := generateTestData(inputSize)
 
+	const mib = 1024 * 1024 * 1
+
 	var baseBuff bytes.Buffer
+	baseBuff.Grow(mib)
```
Comment (lev112) [inline performance/parquet_performance_test.go:81]: not really needed,
the first iteration of the benchmark is discarded, and is used to grow the buffer to the right size: https://github.com/FindHotel/parquet/blob/master/performance/parquet_performance_test.go#L65-L67

File: fields.go [hunk 2]
```diff
@@ -28,8 +29,7 @@ const (
 )
 
 var (
-  buffpool = bytebufferpool.Pool{}
-  compresspool = bytebufferpool.Pool{}
+	buffpool = bytebufferpool.Pool{}
```
Comment (lev112) [inline fields.go:32]: why did you decide to merge the pools?

File: performance/parquet_performance_test.go (hunk 1 above)
Comment (pzartem) [reply performance/parquet_performance_test.go:81]: cool

File: fields.go (hunk 2 above)
Comment (pzartem) [reply fields.go:32]: I'm not sure why we should use a separate one. The data that we compress uses the same buffer so the `input_data` > `compressed`.

In that case, I think we can utilise everything within one pool.

**Pull request merged**